Source/Blazorise/Components/ModalProvider/ModalProvider.razor.cs

```diff
@@ -0,0 +1,182 @@
+        /// Defaults to true.
+        /// </summary>
+        protected virtual bool GetUseModalStructure => ModalProviderOptions?.UseModalStructure ?? UseModalStructure;
+
+        /// <summary>
+        /// If true modal will scroll to top when opened.
+        /// </summary>
+        [Parameter] public bool ScrollToTop { get; set; } = true;
+
+        /// <summary>
+        /// If true modal will scroll to top when opened.
+        /// </summary>
+        protected virtual bool GetScrollToTop => ModalProviderOptions?.ScrollToTop ?? ScrollToTop;
+
+        /// <summary>
+        /// Occurs before the modal is opened.
+        /// </summary>
+        [Parameter] public Func<ModalOpeningEventArgs, Task> Opening { get; set; }
+
+        /// <summary>
+        /// Occurs before the modal is opened.
+        /// </summary>
+        protected virtual Func<ModalOpeningEventArgs, Task> GetOpening => ModalProviderOptions?.Opening ?? Opening;
+
+        /// <summary>
+        /// Occurs before the modal is closed.
+        /// </summary>
+        [Parameter] public Func<ModalClosingEventArgs, Task> Closing { get; set; }
+
+        /// <summary>
+        /// Occurs before the modal is closed.
+        /// </summary>
+        protected virtual Func<ModalClosingEventArgs, Task> GetClosing => ModalProviderOptions?.Closing ?? Closing;
+
+        /// <summary>
+        /// Occurs after the modal has opened.
+        /// </summary>
+        [Parameter] public EventCallback Opened { get; set; }
+
+        /// <summary>
+        /// Occurs after the modal has opened.
+        /// </summary>
+        protected virtual EventCallback GetOpened => ModalProviderOptions?.Opened ?? Opened;
+
+        /// <summary>
+        /// Occurs after the modal has closed.
+        /// </summary>
+        [Parameter] public EventCallback Closed { get; set; }
+
+        /// <summary>
+        /// Occurs before the modal is closed.
+        /// </summary>
+        protected virtual EventCallback GetClosed => ModalProviderOptions?.Closed ?? Closed;
+
+        /// <summary>
+        /// Specifies the backdrop needs to be rendered for this <see cref="Modal"/>.
+        /// </summary>
+        [Parameter] public bool ShowBackdrop { get; set; } = true;
+
+        /// <summary>
+        /// Specifies the backdrop needs to be rendered for this <see cref="Modal"/>.
+        /// </summary>
+        protected virtual bool GetShowBackdrop => ModalProviderOptions?.ShowBackdrop ?? ShowBackdrop;
+
+        /// <summary>
+        /// Gets or sets whether the component has any animations.
+        /// </summary>
+        [Parameter] public bool Animated { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets whether the component has any animations.
+        /// </summary>
+        protected virtual bool GetAnimated => ModalProviderOptions?.Animated ?? Animated;
```

Comment:
It's strange to have a `Get` prefix on properties. Maybe better convert to function in that case. Or change the naming convention.

Reply:
Properties with lambda are pretty much functions xD . 
Suggestions for naming convention? I don't think it's that bad...

Reply:
But I just noticed the doc is wrong "Gets or sets" :D

Reply:
I know that every property is actually a function internally, but it is disguised as a field.
 
You can leave it with `Get` prefix but make a real function: `protected virtual bool GetAnimated() => ModalProviderOptions?.Animated ?? Animated;`